File: clients/ruby/sync/examples/hello_world.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Hello World example - standalone version
#
# This example demonstrates basic execution patterns.
# Shows how to execute code (simulated).
#
# To run:
#     ruby hello_world.rb
#
# Expected output:
#     Status: completed
#     Output: Hello from unsandbox!

# Simulated result (would normally call API)
result = {
  'status' => 'completed',
  'stdout' => "Hello from unsandbox!\n",
  'stderr' => '',
  'exit_code' => 0
}

puts "Status: #{result['status']}"
puts "Output: #{result['stdout'].strip}"

File: clients/rust/sync/examples/hello_world.rs
// PUBLIC DOMAIN - NO LICENSE, NO WARRANTY
//
// Hello World example for unsandbox Rust SDK - Synchronous Version
//
// This example demonstrates basic synchronous execution using the SDK.
// Shows how to execute code from a Rust program using the sync SDK.
//
// To run:
//     export UNSANDBOX_PUBLIC_KEY="your-public-key"
//     export UNSANDBOX_SECRET_KEY="your-secret-key"
//     cargo run --example hello_world
//
// Expected output:
//     Executing code synchronously...
//     Result status: completed
//     Output: Hello from unsandbox!

use un::{execute_code, resolve_credentials, UnsandboxError};

fn main() -> Result<(), UnsandboxError> {
    // The code to execute
    let code = r#"print("Hello from unsandbox!")"#;

    // Resolve credentials from environment or config files
    let creds = resolve_credentials(None, None)?;

    // Execute the code synchronously
    println!("Executing code synchronously...");
    let result = execute_code("python", code, &creds)?;

    // Check the result
    println!("Result status: {}", result.status);
    println!("Output: {}", result.output.trim());

    if result.exit_code != 0 {
        eprintln!("Exit code: {}", result.exit_code);
        std::process::exit(1);
    }

    Ok(())
}
